src/System.Net.Mime/src/System/Net/Mail/BufferBuilder.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Net.Mail
+{
+    internal sealed class BufferBuilder
+    {
+        private byte[] _buffer;
+        private int _offset;
+
+        internal BufferBuilder() : this(256) { }
+        internal BufferBuilder(int initialSize)
+        {
+            _buffer = new byte[initialSize];
+        }
+
+        private void EnsureBuffer(int count)
+        {
+            if (count > _buffer.Length - _offset)
+            {
+                byte[] newBuffer = new byte[((_buffer.Length * 2) > (_buffer.Length + count)) ? (_buffer.Length * 2) : (_buffer.Length + count)];
+                Buffer.BlockCopy(_buffer, 0, newBuffer, 0, _offset);
+                _buffer = newBuffer;
+            }
+        }
+
+        internal void Append(byte value)
+        {
+            EnsureBuffer(1);
+            _buffer[_offset++] = value;
+        }
+
+        internal void Append(byte[] value)
+        {
+            Append(value, 0, value.Length);
+        }
+
+        internal void Append(byte[] value, int offset, int count)
+        {
+            EnsureBuffer(count);
+            Buffer.BlockCopy(value, offset, _buffer, _offset, count);
+            _offset += count;
+        }
+
+        internal void Append(string value)
+        {
+            Append(value, false);
+        }
+
+        internal void Append(string value, bool allowUnicode)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return;
+            }
+            Append(value, 0, value.Length, allowUnicode);
+        }
+
+        internal void Append(string value, int offset, int count, bool allowUnicode)
+        {
+            if (allowUnicode)
+            {
+                byte[] bytes = Encoding.UTF8.GetBytes(value.ToCharArray(), offset, count);
```

Comment:
There's no GetBytes overload that takes a string along with an offset and a count.  I'm going to leave this as-is for now.